packages/navigator/src/browser/open-editors-widget/navigator-open-editors-commands.ts

```diff
@@ -0,0 +1,47 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Command } from '@theia/core/lib/common';
+
+export namespace OpenEditorsCommands {
+    export const CLOSE_ALL_TABS_FROM_TOOLBAR: Command = {
+        id: 'navigator.close.all.editors',
```

Comment:
For the new commands in the file, should we align their `ids` with that of vscode so the plugin-system can directly reference them without the mapping?

Reply:
See note above, I've kept these commands separate from the plugin commands since their implementation is slightly different in both arguments and behavior. I've added added an additional command with VSCode's ID to the `plugin-vscode-commands-contribution` for `workbench.files.saveAllInGroup` which uses a very similar implementation as the existing `workbench.action.closeEditorsInGroup` and ensured that the two now have `labels` so that they can be accessed from the command palette. It's a bit tricky to figure out exactly where some of these things should live, but this was my best attempt at separating responsibilities. I've also added an appropriate utility in the application shell for saving editors which I am using in the `execute` of the toolbar commands.

Let me know if you have suggestions, thanks!